crates/core/component/dex/src/swap_execution.rs

```diff
@@ -13,21 +13,21 @@ pub struct SwapExecution {
 }
 
 impl SwapExecution {
+    /// Returns the price of the latest execution trace.
     pub fn max_price(&self) -> Result<Option<U128x128>> {
-        let Some(aggregate_input )= self.aggregate_input() else {
+        let Some((input, output)) = self.traces.last().and_then(|trace| {
+            let input = trace.first()?;
+            let output = trace.last()?;
+            Some((input, output))
+        }) else {
             return Ok(None)
         };
 
-        let Some(aggregate_output) = self.aggregate_output() else {
-            return Ok(None)
-        };
-
-        let price = U128x128::ratio(aggregate_input, aggregate_output)?;
-
+        let price = U128x128::ratio(input.amount, output.amount)?;
         Ok(Some(price))
     }
 
-    fn aggregate_input(&self) -> Option<Amount> {
+    pub fn aggregate_input(&self) -> Option<Amount> {
```

Comment:
This is just the input field already recorded in the swap execution though right? So we could delete these two methods entirely?

Reply:
yup good catch , i was testing something, ripping them out entirely. fyi, the last few commits aren't showing up yet (github incident).